### DISCUSSION TITLE
How to style a disabled Button?

### DISCUSSION THREAD

**Discussion by ficast** [tamagui/tamagui#1178] [General]
I am struggling to simple change color for a disabled Button, does anyone can help me with that? This is my current code:

`
const TextButtonFrame = styled(ButtonFrame, {
  name: 'TextButtonFrame',
  focusable: true,
  backgroundColor: 'transparent',
  borderWidth: 0,
  outlineOffset: 0,
  outlineWidth: 0,
  hoverStyle: {
    backgroundColor: 'transparent',
    borderRadius: 0,
    borderWidth: 0,
    outlineWidth: 0,

  },
  focusStyle: {
    backgroundColor: 'transparent',
    borderRadius: 0,
    borderWidth: 0,
    outlineWidth: 0,
  },
  variants: {
    active: {
      true: {
        hoverStyle: {
          backgroundColor: 'transparent',
        },
      },
    },
    disabled: {
      true: {
        backgroundColor: 'transparent',
        pointerEvents: 'none',
      },
    },
  },
})

const TextButtonText = styled(ButtonText, {
  height: '$1',
  backgroundColor: 'transparent',
  borderWidth: 0,
  outlineOffset: 0,
  outlineWidth: 0,
  color: '$orange12',
  hoverStyle: {
    backgroundColor: 'transparent',
    borderRadius: 0,
    color: '$orange10',
  },
  focusStyle: {
    backgroundColor: 'transparent',
    borderRadius: 0,
    color: '$orange10',
  },
})

// to capture the custom variant types you define
type CustomButtonFrameProps = GetProps<typeof TextButtonFrame>
type CustomButtonTextProps = GetProps<typeof TextButtonText>

export type CustomButtonProps = TamaguiButtonProps &
  CustomButtonFrameProps &
  CustomButtonTextProps

export const TextButton = TextButtonFrame.styleable<CustomButtonProps>((propsIn, ref) => {
  const { props } = useButton(propsIn, { Text: TextButtonText })
  return <TextButtonFrame {...props} ref={ref} />
})`

**Comment by ardasatata** [comment]
has it resolved?
